crates/ruff_python_formatter/src/expression/binary_like.rs

```diff
@@ -540,7 +633,86 @@ impl Format<PyFormatContext<'_>> for FlatBinaryExpressionSlice<'_> {
     fn fmt(&self, f: &mut Formatter<PyFormatContext>) -> FormatResult<()> {
         // Single operand slice
         if let [OperandOrOperator::Operand(operand)] = &self.0 {
-            return operand.expression().format().fmt(f);
+            let expression = operand.expression();
+
+            return if is_expression_parenthesized(expression.into(), f.context().source()) {
+                let comments = f.context().comments().clone();
+                let expression_comments = comments.leading_dangling_trailing(expression);
+
+                // Format leading comments that are before any `(` outside of the expression-parentheses.
```

Comment:
Does rposition still do the right thing with multiple parentheses?

Reply:
I added a new test to `binary.py`. Whether it is subjective: 

```
(
               # comment
               (1 << (n + i-j + n-1))) # NW-SE ordinal
```

The `# comment` gets moved out of the parentheses because the formatter removes the unnecessary outer most parentheses. 

```python
    # comment
    (1 << (n + i - j + n - 1))  # NW-SE ordinal
```

I think that's fine and the result is stable.